CommonTools/ParticleFlow/plugins/PFNoPileUpPacked.cc

```diff
@@ -0,0 +1,91 @@
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+using namespace std;
+using namespace edm;
+using namespace reco;
+
+/**\class PFNoPileUpPacked
+\brief Identifies pile-up candidates from a collection of Candidates, and
+produces the corresponding collection of NoPileUpCandidates.
+
+\author Andreas Hinzmann
+\date   May 2021
+
+*/
+
+class PFNoPileUpPacked : public edm::stream::EDProducer<> {
+public:
+  typedef edm::View<reco::Candidate> CandidateView;
+  typedef edm::Association<reco::VertexCollection> CandToVertex;
+
+  explicit PFNoPileUpPacked(const edm::ParameterSet&);
+
+  ~PFNoPileUpPacked() override = default;
+
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  edm::EDGetTokenT<CandidateView> tokenCandidatesView_;
+  edm::EDGetTokenT<reco::VertexCollection> tokenVertices_;
+  edm::EDGetTokenT<CandToVertex> tokenVertexAssociation_;
+  edm::EDGetTokenT<edm::ValueMap<int>> tokenVertexAssociationQuality_;
+  int vertexAssociationQuality_;
+};
+
+PFNoPileUpPacked::PFNoPileUpPacked(const edm::ParameterSet& iConfig) {
+  tokenCandidatesView_ = consumes<CandidateView>(iConfig.getParameter<InputTag>("candidates"));
+  vertexAssociationQuality_ = iConfig.getParameter<int>("vertexAssociationQuality");
+  tokenVertexAssociation_ = consumes<CandToVertex>(iConfig.getParameter<edm::InputTag>("vertexAssociation"));
+  tokenVertexAssociationQuality_ =
+      consumes<edm::ValueMap<int>>(iConfig.getParameter<edm::InputTag>("vertexAssociation"));
+  produces<edm::PtrVector<reco::Candidate>>();
+}
+
+void PFNoPileUpPacked::produce(Event& iEvent, const EventSetup& iSetup) {
+  unique_ptr<edm::PtrVector<reco::Candidate>> pOutput(new edm::PtrVector<reco::Candidate>);
+  Handle<CandidateView> candidateView;
+  iEvent.getByToken(tokenCandidatesView_, candidateView);
+  const edm::Association<reco::VertexCollection>& associatedPV = iEvent.get(tokenVertexAssociation_);
+  const edm::ValueMap<int>& associationQuality = iEvent.get(tokenVertexAssociationQuality_);
+  for (const auto& p : candidateView->ptrs()) {
+    const reco::VertexRef& PVOrig = associatedPV[p];
+    int quality = associationQuality[p];
+    if (!(PVOrig.isNonnull() && (PVOrig.key() > 0) && (quality >= vertexAssociationQuality_)))
+      pOutput->push_back(p);
+  }
+  iEvent.put(std::move(pOutput));
+}
+
+void PFNoPileUpPacked::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("candidates", edm::InputTag("packedPFCandidates"));
+  desc.add<int>("vertexAssociationQuality", 7);
+  desc.add<edm::InputTag>("vertexAssociation", edm::InputTag("packedPrimaryVertexAssociationJME", "original"));
+  descriptions.add("produceMuons", desc);
```

Comment:
> There is no "CommonTools.ParticleFlow.pfNoPileUpPacked_cfi" or "CommonTools.ParticleFlow.pFNoPileUpPacked_cfi" python config being generated, though there is a fillDescriptions defined and a DEFINE_FWK_MODULE(PFNoPileUpPacked). What's missing to have it generated?

This call leads to generation of `produceMuons_cfi.py`. Either use different label, or use
```suggestion
  descriptions.addWithDefaultLabel(desc);
```
for auto-generated label.

Reply:
thanks, this escaped my eyes